test/e2e/delete_test.go

```diff
@@ -23,42 +22,45 @@ func testWindowsNodeDeletion(t *testing.T) {
-	var windowsMachineSetWithLabel string
-	var e2eMachineSets []*mapi.MachineSet
-	for _, machineSet := range machineSetList.Items {
-		if strings.Contains(machineSet.Name, "e2e-windows-machineset-") {
-			e2eMachineSets = append(e2eMachineSets, &machineSet)
-		}
-		if machineSet.Spec.Selector.MatchLabels["machine.openshift.io/os-id"] == "Windows" {
-			windowsMachineSetWithLabel = machineSet.Name
+	var windowsMachineSetWithLabel *mapi.MachineSet
+	for _, machineSet := range e2eMachineSets.Items {
+		if machineSet.Spec.Selector.MatchLabels[clusterinfo.MachineOSIDLabel] == "Windows" {
+			windowsMachineSetWithLabel = &machineSet
+			break
 		}
 	}
-	windowsMachineSet, err := testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Get(context.TODO(),
-		windowsMachineSetWithLabel, meta.GetOptions{})
-	require.NoError(t, err, "error getting Windows MachineSet")
-	// Reset the number of nodes to be deleted to 0
-	gc.numberOfNodes = 0
-	// Delete the Windows VM that got created.
-	windowsMachineSet.Spec.Replicas = &gc.numberOfNodes
-	_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(), windowsMachineSet,
-		meta.UpdateOptions{})
-	require.NoError(t, err, "error updating Windows MachineSet")
-	// we are waiting 10 minutes for all windows machines to get deleted.
-	err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
-	require.NoError(t, err, "Windows node deletion failed")
 
-	// Cleanup all the MachineSets created by us.
-	for _, machineSet := range e2eMachineSets {
-		assert.NoError(t, testCtx.deleteMachineSet(machineSet), "error deleting MachineSet")
-	}
-	// Phase is ignored during deletion, in this case we are just waiting for Machines to be deleted.
-	require.NoError(t, testCtx.waitForWindowsMachines(int(gc.numberOfNodes), ""), "Windows machine deletion failed")
+	t.Run("Windows Nodes scale down with Machines", func(t *testing.T) {
+		require.NotNilf(t, windowsMachineSetWithLabel, "could not find MachineSet with Windows label")
+		// Set the expected number of Windows nodes to 0
+		gc.numberOfNodes = 0
+
+		// Scale the Windows MachineSet to 0
+		windowsMachineSetWithLabel.Spec.Replicas = &gc.numberOfNodes
+		_, err = testCtx.client.Machine.MachineSets(clusterinfo.MachineAPINamespace).Update(context.TODO(),
+			windowsMachineSetWithLabel, meta.UpdateOptions{})
+		require.NoError(t, err, "error updating Windows MachineSet")
+
+		// we are waiting 10 minutes for all windows machines to get deleted.
+		err = testCtx.waitForWindowsNodes(gc.numberOfNodes, true, true, false)
+		require.NoError(t, err, "Windows node deletion failed")
+	})
+
+	t.Run("Prometheus configuration is updated when Machines are deleted", func(t *testing.T) {
+		// Cleanup all the MachineSets created by us.
+		for _, machineSet := range e2eMachineSets.Items {
+			assert.NoError(t, testCtx.deleteMachineSet(&machineSet), "error deleting MachineSet")
+		}
```

Comment:
Why are you doing this as part of this test? This should happen irrespective or am I missing something?